osquery/tables/system/darwin/keychain_acl.cpp

```diff
@@ -0,0 +1,334 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <sstream>
+
+#include <Security/Security.h>
+
+#include <boost/format.hpp>
+
+#include <osquery/core.h>
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/sql.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+#include "osquery/tables/system/darwin/keychain.h"
+
+namespace osquery {
+namespace tables {
+
+typedef struct {
+  std::string keychain_path;
+  std::string label;
+} KeychainItemMetadata;
+
+typedef struct {
+  std::vector<std::string> authorizations;
+  std::string description;
+  std::vector<std::string> applications;
+} KeychainItemACL;
+
+const std::map<CSSM_ACL_AUTHORIZATION_TAG, std::string> kACLAuthorizationTags =
+    {
+        {CSSM_ACL_AUTHORIZATION_ANY, "any"},
+        {CSSM_ACL_AUTHORIZATION_LOGIN, "login"},
+        {CSSM_ACL_AUTHORIZATION_GENKEY, "genkey"},
+        {CSSM_ACL_AUTHORIZATION_DELETE, "delete"},
+        {CSSM_ACL_AUTHORIZATION_EXPORT_WRAPPED, "export_wrapped"},
+        {CSSM_ACL_AUTHORIZATION_EXPORT_CLEAR, "export_clear"},
+        {CSSM_ACL_AUTHORIZATION_IMPORT_WRAPPED, "import_wrapped"},
+        {CSSM_ACL_AUTHORIZATION_IMPORT_CLEAR, "import_clear"},
+        {CSSM_ACL_AUTHORIZATION_SIGN, "sign"},
+        {CSSM_ACL_AUTHORIZATION_ENCRYPT, "encrypt"},
+        {CSSM_ACL_AUTHORIZATION_DECRYPT, "decrypt"},
+        {CSSM_ACL_AUTHORIZATION_MAC, "mac"},
+        {CSSM_ACL_AUTHORIZATION_DERIVE, "derive"},
+        {CSSM_ACL_AUTHORIZATION_DBS_CREATE, "dbs_create"},
+        {CSSM_ACL_AUTHORIZATION_DBS_DELETE, "dbs_delete"},
+        {CSSM_ACL_AUTHORIZATION_DB_READ, "db_read"},
+        {CSSM_ACL_AUTHORIZATION_DB_INSERT, "db_insert"},
+        {CSSM_ACL_AUTHORIZATION_DB_MODIFY, "db_modify"},
+        {CSSM_ACL_AUTHORIZATION_DB_DELETE, "db_delete"},
+        {CSSM_ACL_AUTHORIZATION_CHANGE_ACL, "change_acl"},
+        {CSSM_ACL_AUTHORIZATION_CHANGE_OWNER, "change_owner"},
+};
+
+Status parseKeychainItemACL(SecAccessRef access,
+                            std::vector<KeychainItemACL> &acls) {
+  OSStatus os_status;
+  CFArrayRef acl_list = nullptr;
+  os_status = SecAccessCopyACLList(access, &acl_list);
```

Comment:
i'm not going to do auto, since i re-use os_status rather often in this method, i'd rather it's definition be explicit, since it could easily be confused with osquery::Status. i'll see if i can find the macro for OSStatus 0.